### PR TITLE
[17.0][MIG] endpoint: Migration to 17.0

### PR DESCRIPTION
**Pull request by SilvioC2C**

### CODE REVIEW

**Review by yankinmax — APPROVED** [review]

File: endpoint/models/endpoint_mixin.py [hunk 1]
```diff
@@ -206,10 +229,9 @@ def _find_endpoint_domain(self, endpoint_route):
...
-        result = super().copy_data(default=default)
-        # `route` cannot be copied as it must me unique.
+        # OVERRIDE: ``route`` cannot be copied as it must me unique.
         # Yet, we want to be able to duplicate a record from the UI.
-        for rec, data in zip(self, result):
-            if not data.get("route"):
-                data["route"] = f"{rec.route}/COPY_FIXME"
-        return result
+        self.ensure_one()
+        default = dict(default or {})
```
Comment (simahawk) [inline endpoint/models/endpoint_mixin.py:235]: suggestion
```
        default = default or {}

**Review by simahawk — APPROVED** [review]
LG minor remark

File: endpoint/models/endpoint_mixin.py (hunk 1 above)
Comment (SilvioC2C) [reply endpoint/models/endpoint_mixin.py:235]: This is how Odoo handles it in the `super()` method, I preferred to keep it consistent

**Comment by SilvioC2C** [comment]
@simahawk shall we merge? It will unlock https://github.com/OCA/edi-framework/pull/124 :smile:

**Pull request merged**